Classes/Form/Field/Text.php

```diff
@@ -202,13 +200,8 @@ public function getRichtextConfiguration()
      */
     protected function getPageTsConfigForRichTextEditor()
     {
-        $configurationManager = $this->getObjectManager()->get(ConfigurationManagerInterface::class);
-        if ($configurationManager instanceof BackendConfigurationManager) {
-            $pageUid = $configurationManager->getCurrentPageId();
-        } else {
-            $root = $this->getRoot();
-            $pageUid = $root instanceof Form ? $root->getOption('record')['pid'] ?? 0 : 0;
-        }
+        $root = $this->getRoot();
+        $pageUid = $root instanceof Form ? $root->getOption('record')['pid'] : 0;
```

Comment:
This has to be`uid` instead of `pid`, since we're already looking at a page. I have the Page TSConfig defined on page 1 (site root), and with `pid` it will look for the TSConfig on page 0 (tree root) when editing the site root page, which is wrong.

Reply:
Wait, this also applies to flexforms in content elements, right? So it could be both depending on the context... :-)

Reply:
Yes, it could be both depending on the context...
In most cases it is a content element, but yes it could be a Page Configuration as richtext too.